functionalReduce.js

```diff
@@ -0,0 +1,14 @@
+function reduce (head, f, init){
+  var accumulated = init
+  var current = head
+
+  if(current === null){return f(accumulated,head)}
```

Comment:
I prefer to use multi-line blocks in order to make code easier to read and maintain.  The following is a little easier for my brain to parse, for example:
```js
if( current === null ) {
  return f( accumulated, head )
}
```